src/mono/mono/mini/aot-compiler.c

```diff
@@ -9864,9 +9864,9 @@ compile_method (MonoAotCompile *acfg, MonoMethod *method)
 
 	mono_atomic_inc_i32 (&acfg->stats.ccount);
 
-	if (acfg->aot_opts.compiled_methods_outfile && acfg->compiled_methods_outfile != NULL) {
-		if (!mono_method_is_generic_impl (method) && method->token != 0) {
-			fprintf (acfg->compiled_methods_outfile, "%x\n", method->token);
+	if (acfg->aot_opts.trimming_eligible_methods_outfile && acfg->trimming_eligible_methods_outfile != NULL) {
+		if (!mono_method_is_generic_impl (method) && method->token != 0 && !cfg->deopt && !cfg->interp_entry_only && mini_get_interp_callbacks ()->jit_call_can_be_supported (method, mono_method_signature_internal (method))) {
```

Comment:
does jit_call_can_be_supported always return the same thing for host (x86 / x64 / arm) and target (wasm)? are we sure that will remain true in the future? (a comment near that function specifying that we rely on this would be sufficient, i think, if one isn't already there)

Reply:
One thing worth mentioning here is that `jit_call_can_be_supported` always return TRUE when `interpreter` is not being used. Thus, condition doesn't affect targets which doesn't use interpreter, such as Android.